### PR TITLE
 Fix scoping bug for errors on collector service Run

### PR DESCRIPTION
**Pull request by phenixrizen**
I had a tab error in my config yaml and the extension layer collector was hanging with no error output which led me to this bug.

The current logic will never receive the error returned from `appErr := c.svc.Run(ctx)` because error will be out of scope and the collector will hang. This creates a channel and assures we check the error. This allows us to get the config error.

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -77,13 +77,18 @@ func (c *Collector) Start(ctx context.Context) error {
...
-	go func() {
+	runErr := make(chan error, 1)
+	go func(runErr chan error) {
 		defer close(c.appDone)
-		appErr := c.svc.Run(ctx)
-		if appErr != nil {
-			err = appErr
-		}
-	}()
+		err := c.svc.Run(context.Background())
```
Comment (NathanielRN) [inline collector/collector.go]: Just curious, why is `context.Background()` better than `ctx` here?

**Review by NathanielRN — APPROVED** [review]
Thanks for this change! It looks useful 🙂

File: collector/collector.go (hunk 1 above)
Comment (phenixrizen) [reply collector/collector.go]: Good catch that should be the passed context. I was working off a older version that had no context passed to Run.

**Comment by phenixrizen** [comment]
Any update on getting this merged would like to use this instead of a fork.

**Comment by NathanielRN** [comment]
It's hard with people on holidays I guess, but I also would like to see this merged because I have ran into it too 😞

@codeboten think you could merge this when you have a chance? 🙂

**Review by codeboten — APPROVED** [review]
Thanks for fixing!

**Pull request merged**